app/assets/javascripts/controllers/middleware_server/middleware_server_controller.js

```diff
@@ -171,20 +181,30 @@ function MwServerOpsController( miqService, serverOpsService) {
 }
 
 ManageIQ.angular.app.service('serverOpsService', ServerOpsService);
+ManageIQ.angular.app.service('serverGroupOpsService', ServerGroupOpsService);
 
 ServerOpsService.$inject = ['$http', '$q'];
-
 function ServerOpsService($http, $q) {
-  this.runOperation = function runOperation(id, operation, timeout) {
-    var errorMsg = _('Error running operation on this server.');
+  return ServerOpsServiceFactory($http, $q, false);
+}
+
+ServerGroupOpsService.$inject = ['$http', '$q'];
+function ServerGroupOpsService($http, $q) {
+  return ServerOpsServiceFactory($http, $q, true);
+}
+
+function ServerOpsServiceFactory($http, $q, isGroup) {
+  var runOperation = function runOperation(id, operation, timeout) {
+    var errorMsg = _('Error running operation on this server' + (isGroup ? '.' : ' group.'));
```

Comment:
Please, don't do math on strings that need translation. It's better to repeat the common substring rather than making the hell from the translation work ;-)